### ISSUE TITLE
JAWS leaves forms mode after pressing enter key in an inline CKEditor instance

### ISSUE THREAD

**Issue by jswiderski** [ckeditor/ckeditor-dev#1034]
## Are you reporting a feature request or a bug?

Bug

## Provide detailed reproduction steps (if any)

0. **AUTO FORMS MODE** - According to this link - https://tink.uk/understanding-screen-reader-interaction-modes/, forms mode is being activated automatically whenever you enter a form filed. It is done automatically and signalized with a beep.
**MANUAL FORMS MODE** - According to this link http://doccenter.freedomscientific.com/doccenter/doccenter/rs11f929e9c511/2014-12-02_webpagetesting-l1/02_JAWS-Settings.htm you can enable Manual Forms Mode by pressing `INSERT+F2`, selecting Settings Center, typing Forms Mode and switching from Auto to Manual. To enable Forms mode when on page, you need to press Enter. To disable it, you need to press `Esc`.

1. Enable JAWS
2. Open inline editor sample and focus editor (There will be a beep signalizing entering the forms mode)
3. Clear editor contents, type few characters and press enter

### Expected result

While editor is focused and Forms Mode hasn't been disabled with `Esc` (only in Manual Forms Mode), the Forms Mode should be on.

### Actual result

In Inline editor Forms Mode gets disabled with every enter key pressed and you actually need to press enter twice with every new line to be able to still type in the editor. 

If you try the same steps with native `contenteditable` element e.g. https://codepen.io/j_swiderski/pen/NvOdWz, you will see it works fine there.

## Other details

* Browser: Firefox
* OS: Windows
* CKEditor version: 4.0+
* Installed CKEditor plugins: N/A

**Comment by mlewand** [comment]
I can no longer reproduce this issue, either for older CKEditor (e.g. 4.6.0) as well for the latest CKEditor.

I have used the old "Replace Textarea with Inline Editor" as it contains an inline editor in a `form` element.

Repro steps are:

1. Open the old ["Replace Textarea with Inline Editor" sample](http://nightly.ckeditor.com/latest/standard/ckeditor/samples/old/inlinetextarea.html).
1. Navigate to the editor.
1. Type "h" few times.
1. Press "Enter" twice.
1. Type "h" few times.

## Expected & Actual

"h" character is typed into the editable. If bug was present it would move to a next header.

Below a screencast:

* [double enter latest CKEditor](https://github.com/ckeditor/ckeditor-dev/files/1770700/gh1034.-.double.enter.latest.CKE.zip)
* [double enter on CKEditor 4.6.0](https://github.com/ckeditor/ckeditor-dev/files/1770702/gh1034.-.double.enter.460.zip)

JAWS version: 2018.1802.78
Firefox version: 58.0.2
OS: Windows 7

**Comment by mlewand** [comment]
I was able to reproduce it though with Firefox ESR 52.6.0 and JAWS 18.0.2945. However we'd like to hear from you if the issue is still reproducible with the latest JAWS version.

**Comment by mlewand** [comment]
I was able to set up latest JAWS (it was PITA). The source of the problem is explained in #1780.

**Comment by Comandeer** [comment]
Closed in #1780.

**Comment by mlewand** [comment]
Issue has been reviewed. It landed to our upcoming 4.10.0 release, developed in `next` branch (currently there's no ETA, as soon as we have it it will be posted on https://github.com/ckeditor/ckeditor-dev/milestones).

Meanwhile you could have a look at it on our online preview page: https://ckeditor4-preview.ckeditor.com/next/samples/old/inlinetextarea.html

**Comment by jswiderski** [comment]
I also confirm that adding `role=textbox` to native editable element just like in CKEditor, breaks Forms Mode and adding the `aria-multiline="true"` fixes it.

**Comment by JulesAchiel** [comment]
Is this issue fixed in 4.10 ?

**Comment by mlewand** [comment]
@JulesAchiel yes it is, please have a look at https://github.com/ckeditor/ckeditor-dev/issues/1034#issuecomment-372368160.
